### DISCUSSION TITLE
OCR Coordinates do not match

### DISCUSSION THREAD

**Discussion by xiaolibuzai-ovo** [pymupdf/PyMuPDF#4069] [Looking for help]
### Description of the bug

I used another OCR to recognize the content coordinates of the PDF, and then I used the PyMuPDF library. I hope to extract the coordinates of a specified area, but there is a significant difference between the two sets of coordinates.

These are the coordinates recognized by the other OCR:
{
        "text": "Vue Mastery",
        "bbox": [
            586.0,
            178.0,
            1250.0,
            296.0
        ],
        "type": "ocr",
        "score": 1
    }
These are the coordinates for the corresponding position in PyMuPDF:
(88.85449981689453, 23.943227767944336, 117.37201690673828, 44.796356201171875, 'Vue', 0, 0, 0), (121.81803131103516, 23.943227767944336, 183.36544799804688, 44.796356201171875, 'Mastery', 0, 0, 1)

this is pdf file
[Nuxtjs-Cheat-Sheet.pdf](https://github.com/user-attachments/files/17828390/Nuxtjs-Cheat-Sheet.pdf)

### How to reproduce the bug

see above

Hope to be answered

### PyMuPDF version

1.24.14

### Operating system

MacOS

### Python version

3.10

**Comment by JorjMcKie** [comment]
Look at this code:

![image](https://github.com/user-attachments/assets/a02cb319-fc44-449d-9b81-13835f220c9d)

Looks like the PyMuPDF coordinates are correct. That mysterious "other OCR" tool provides coordinates outside the dimension of an A4 page: x-values should not exceed 596, but we see a value 1250.

I also do not understand why we talk about OCR at all: the text can be extracted with no problem, and none of the 42 images covers the page.

**Reply by xiaolibuzai-ovo** [reply]
> Look at this code:
> 
> ![image](https://private-user-images.githubusercontent.com/8290722/388046474-a02cb319-fc44-449d-9b81-13835f220c9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDI2NTUsIm5iZiI6MTczMjEwMjM1NSwicGF0aCI6Ii84MjkwNzIyLzM4ODA0NjQ3NC1hMDJjYjMxOS1mYzQ0LTQ0OWQtOWI4MS0xMzgzNWYyMjBjOWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMTEzMjM1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTQ2MjNiN2FiZWVlZjljNGQ0OWI0MjcwMDRkNTAxYWNjNjRkMjlkZjY2OTM0NmVlNWZhZDBjOTM2ODE0ZGQzZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.AzlZX3G-39v7gfwdU-bNJrc2vYog1arCu0BQBcH2R8g)
> 
> Looks like the PyMuPDF coordinates are correct. That mysterious "other OCR" tool provides coordinates outside the dimension of an A4 page: x-values should not exceed 596, but we see a value 1250.
> 
> I also do not understand why we talk about OCR at all: the text can be extracted with no problem, and none of the 42 images covers the page.

I used the code you provided to parse the PDF file mentioned above, but it seems that I have lost the script tags？
<img width="1512" alt="image" src="https://github.com/user-attachments/assets/62c221ab-93f4-4054-b9e7-c7dc75bc1a1f">

**Comment by xiaolibuzai-ovo** [comment]
> Look at this code: 看这段代码：
> 
> ![image](https://private-user-images.githubusercontent.com/8290722/388046474-a02cb319-fc44-449d-9b81-13835f220c9d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwOTg1MjgsIm5iZiI6MTczMjA5ODIyOCwicGF0aCI6Ii84MjkwNzIyLzM4ODA0NjQ3NC1hMDJjYjMxOS1mYzQ0LTQ0OWQtOWI4MS0xMzgzNWYyMjBjOWQucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMTAyMzQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjVmZWVjZTUxMDIyMDkyOGYyMDQ1NjFkM2E1ODE5NGE4MDQxOGU0ZmEyNzFmYTgzNDhhNjcyMzc1YjllMjUwMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.DbZa-sAu4Vn1dDXrEzoivWPOzp2c0hseHEcq65qnH-Y)
> 
> Looks like the PyMuPDF coordinates are correct. That mysterious "other OCR" tool provides coordinates outside the dimension of an A4 page: x-values should not exceed 596, but we see a value 1250.看起来 PyMuPDF 坐标是正确的。这个神秘的“其他 OCR”工具提供 A4 页面尺寸之外的坐标：x 值不应超过 596，但我们看到的值是 1250。
> 
> I also do not understand why we talk about OCR at all: the text can be extracted with no problem, and none of the 42 images covers the page.我也不明白为什么我们要谈论OCR：文本可以毫无问题地提取，并且42张图像没有一张覆盖页面。

Thank you for your reply. I am currently facing this issue: using PyMuPDF to recognize the PDF leads to inaccuracies in the content recognition. For example, I used the translation script from https://github.com/pymupdf/PyMuPDF-Utilities/blob/tutorials/tutorials/language-translation/translator.py, but the restored content differs significantly. Here is my PDF:
[Nuxtjs-Cheat-Sheet.pdf](https://github.com/user-attachments/files/17829005/Nuxtjs-Cheat-Sheet.pdf)
result：
<img width="1854" alt="企业微信截图_2c1fd6cc-dc5e-4671-b16c-7f4580a39388" src="https://github.com/user-attachments/assets/af5707ce-41a4-4fe3-ab22-ec477c21a2a5">

I can accurately select the positions using another OCR, so my idea is to have the OCR find the positions, and then translate and write back by extracting the matrix content.

other OCR result：
![vue pdf_0](https://github.com/user-attachments/assets/5154e123-3f81-4d1c-a0d7-1e8e80d053b0)

**Comment by JorjMcKie** [comment]
Still don't understand why we even talk about OCR.
PyMuPDF can correctly detect all text **_natively_** without any problem and top precision:
![image](https://github.com/user-attachments/assets/c4c33083-21cf-4ac8-8cd9-d2b2aa98987a)

I think we are talking past each other: 
* **_What is it that you are really trying to do?_**

Language translation of a given document maybe?

**Reply by xiaolibuzai-ovo** [reply]
> Still don't understand why we even talk about OCR.仍然不明白为什么我们还要谈论 OCR。 PyMuPDF can correctly detect all text **_natively_** without any problem and top precision:PyMuPDF 可以正确地**_原生_**检测所有文本，没有任何问题并且精度很高： ![image](https://private-user-images.githubusercontent.com/8290722/388058702-c4c33083-21cf-4ac8-8cd9-d2b2aa98987a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDAwODIsIm5iZiI6MTczMjA5OTc4MiwicGF0aCI6Ii84MjkwNzIyLzM4ODA1ODcwMi1jNGMzMzA4My0yMWNmLTRhYzgtOGNkOS1kMmIyYWE5ODk4N2EucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTEyMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMjBUMTA0OTQyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NTFhZTU1MzgyOWMzZjA3N2VhMzBjMTIyZjJiYzkyNjJkNzA0MDZlYjQyNzliZjNhNGRlZDVlNDVlYTdkMzU5YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.PCqnDLBGg6VOQqW1-bMXnyggFbXVYmU-MiVK_sxJCxw)
> 
> I think we are talking past each other:我认为我们正在谈论彼此：
> 
> * **_What is it that you are really trying to do?你真正想做的是什么？_**
> 
> Language translation of a given document maybe?也许给定文档的语言翻译？

you are right! this is a Discussions, I'm sry!

Is this image the one you recognized? Can I take a look at the code? It seems that there are some differences compared to the results from the translation script I used (in the image I provided above, there are many areas that were not recognized by the translation script).

**Comment by JorjMcKie** [comment]
I am transferring this post to "Discussions" as we are clearly not dealing with a bug.
